pages/verification/index.tsx

```diff
@@ -61,33 +62,39 @@ export default function Verification(props: PageProps) {
         apiAnonymous.defaults.headers.common['Authorization'] = `Bearer ${anonymousToken}`;
         if(withLoading) setIsLoading(true);
         generateAnonUser.mutate({ userId: member }, {
-            onSuccess: (data) => {
-                initChatAnon.mutate({
-                    anon_user_info_color_code: data.anon_user_info_color_code,
-                    anon_user_info_color_name: data.anon_user_info_color_name,
-                    anon_user_info_emoji_code: data.anon_user_info_emoji_code,
-                    anon_user_info_emoji_name: data.anon_user_info_emoji_name,
-                    member: member,
-                    message: message,
-                }, {
-                    onSuccess: (data) => {
-                        if (data) {
-                            if(replaceRoute) router.replace('/message-sent');
-                            else router.push('/message-sent');
-                        }
+            onSuccess: async (data) => {
+                try {
+                    const response = await sendMessage({
+                        anon_user_info_color_code: data.anon_user_info_color_code,
+                        anon_user_info_color_name: data.anon_user_info_color_name,
+                        anon_user_info_emoji_code: data.anon_user_info_emoji_code,
+                        anon_user_info_emoji_name: data.anon_user_info_emoji_name,
+                        member: member,
+                        message: message,
+                    })
+    
+                    if(response?.success) {
+                        if(replaceRoute) router.replace('/message-sent');
+                        else router.push('/message-sent');
                         localStorage.removeItem(MessageEnum.tempMessage);
                         localStorage.removeItem(MessageEnum.targetUser);
                         setIsLoading(false);
-                    },
-                    onError: (err) => {
+                    } else {
                         setIsLoading(false);
-                        console.error(err)
+                        console.error(response?.message)
                         toast('We failed to send your message', {
                             autoClose: 3000,
                             type: 'error',
                         })
                     }
-                })
+                } catch(e) {
+                    setIsLoading(false);
+                    console.error(e)
+                    toast('We failed to send your message', {
+                        autoClose: 3000,
+                        type: 'error',
+                    })
+                }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error handling in this section could be improved. Currently, if `sendMessage` fails (either by throwing an exception or returning a response with `success: false`), the same error message is displayed to the user: 'We failed to send your message'. This doesn't provide much information about what went wrong.

Consider differentiating between these two types of errors and providing more specific error messages. For example, if `sendMessage` throws an exception, it might be due to a network issue, while if it returns `success: false`, there might be a problem with the server or the request parameters.

Also, you are setting `setIsLoading(false)` in both the success and error cases. To avoid repetition, consider moving this to a `finally` block.

```diff
            onSuccess: async (data) => {
                try {
                    const response = await sendMessage({
                        anon_user_info_color_code: data.anon_user_info_color_code,
                        anon_user_info_color_name: data.anon_user_info_color_name,
                        anon_user_info_emoji_code: data.anon_user_info_emoji_code,
                        anon_user_info_emoji_name: data.anon_user_info_emoji_name,
                        member: member,
                        message: message,
                    })
    
                    if(response?.success) {
                        if(replaceRoute) router.replace('/message-sent');
                        else router.push('/message-sent');
                         localStorage.removeItem(MessageEnum.tempMessage);
                         localStorage.removeItem(MessageEnum.targetUser);
                    } else {
                        console.error(response?.message)
                         toast('Server error: Failed to send your message', {
                             autoClose: 3000,
                             type: 'error',
                         })
                     }
                } catch(e) {
                    console.error(e)
                    toast('Network error: Failed to send your message', {
                        autoClose: 3000,
                        type: 'error',
                    })
                } finally {
                    setIsLoading(false);
                }
            },
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->